containers/container_implementation.md

```diff
@@ -2,24 +2,23 @@
 The following is relevant for the implementation of model containers in different languages. 
 Model containers must implement functionality consistent with this information.
 
-## Socket Creation 
-- Create a [ZeroMQ Dealer socket](http://api.zeromq.org/3-2:zmq-socket). This socket can receive requests and send responses asynchronously.
-  * Python example:
-  
-    ```
-    import zmq
-    context = zmq.Context();
-    socket = context.socket(zmq.DEALER)
-    ```
+## Connection Lifecycle
+The connection lifecycle defines the socket construction, destruction, and state-based behavior for a single Clipper session. We define a session as a sustained connection between a container and Clipper. Key points:
 
-## Initializing a Connection
-1. Once a ZeroMQ Dealer socket has been created, use it to connect to Clipper. In Python, this can be accomplished as follows:
+- Every session requires a distinct socket. When a session ends, a container must create a new socket in order to reconnect to Clipper.
+- Sessions are initiated and sustained by a heartbeating process. The initial detection of a two-way heartbeat between Clipper and a container marks the creation of a successful connection, and the loss of heartbeat indicates that a connection is broken.

```

Comment:
Note whether the heartbeat is sent from Clipper to the container or the container to Clipper